network-libp2p/src/network.rs

```diff
@@ -686,21 +686,46 @@ impl Network {
                 match event {
                     NimiqEvent::Dht(event) => {
                         match event {
-                            KademliaEvent::OutboundQueryProgressed { id, result, .. } => {
+                            KademliaEvent::OutboundQueryProgressed {
+                                id,
+                                result,
+                                stats: _,
+                                step,
+                            } => {
                                 match result {
-                                    QueryResult::GetRecord(result) => {
+                                    QueryResult::GetRecord(Ok(GetRecordOk::FoundRecord(
+                                        record,
+                                    ))) => {
                                         if let Some(output) = state.dht_gets.remove(&id) {
-                                            let result = result
-                                                .map_err(Into::into)
-                                                .map(|record| match record {
-                                                    GetRecordOk::FoundRecord(r) => Some(r.record.value),
-                                                    GetRecordOk::FinishedWithNoAdditionalRecord { cache_candidates: _ } => None,
-                                                });
-                                            if output.send(result).is_err() {
+                                            // Finish the query. We are only interested in the first result.
+                                            // TODO: Revisit this since we are using a Quorum of 1 to report the
+                                            // record to the application. We may want more but also need a way
+                                            // to verify and select the bests record.
+                                            swarm
+                                                .behaviour_mut()
+                                                .dht
+                                                .query_mut(&id)
+                                                .unwrap()
+                                                .finish();
+                                            if output.send(Ok(record.record.value)).is_err() {
                                                 error!(query_id = ?id, error = "receiver hung up", "could not send get record query result to channel");
                                             }
                                         } else {
-                                            warn!(query_id = ?id, "GetRecord query result for unknown query ID");
+                                            warn!(query_id = ?id, ?step, "GetRecord query result for unknown query ID");
```

Comment:
Is this in fact still a warn, or expected behaviour?

Reply:
It would be unexpected, thus the warning